subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/AbstractReportTask.java

```diff
@@ -71,6 +72,10 @@ public void generate() {
             @Override
             public void generateReport(Project project) throws IOException {
                 generate(project);
+
+                if (!outputsToConsole()) {
```

Comment:
Could also  reverse the condition and name the private method to outputsToFile() instead.